### PR TITLE
Add build alias

### PR DESCRIPTION
**Pull request by zklaus**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

This implements the deduplication strategy discussed in conda-forge/build-feedstock#2.

### CODE REVIEW

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -10,32 +10,54 @@ source:
...
+      imports:
+        - build
+      commands:
+        - pip check
+        - python-build --help
+      requires:
+        - pip
+
+
+  - name: build
```
Comment (ocefpaf) [inline recipe/meta.yaml:50]: The reason to name it python-build was b/c the build name was too generic and we did not wanted to take that namespace. Pinging @isuruf here who suggested this in the first place. I have to confess that I don't have a strong opinion but I'd rather renamed it to `build` than maintain the alias.

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by CAM-Gerlach** [comment]
FWIW, as a user of both `build`'s CLI and Python API (as a Pyroma maintainer and others), member of both the PyPA and Conda packaging communities, heavy Conda user, maintainer of a handful of CF recipes and someone who's installed/updated build countless times from both PyPI and CF as well as written the developer/release documentation and CI workflows for a number of projects that all use `build`, I hadn't realized that there was a `python-build` package and `build` was archived until I went and manually checked the other feedstock just now.

Given how there no easy way for users to discover the other name's deprecation status, it seems highly probable that many other users will continue to make this mistake for years to come, if not forever, silently leaving them with an increasingly antiquated version of the `build` package. Therefore, it would seem prudent to either merge this, or actively explore another viable long-term solution.

**Comment by henryiii** [comment]
I also went to build-feedstock from https://conda-forge.org/feedstock-outputs/, and found the "update to 0.8" PR before I realized it was archived and replaced by this one. Since the name is already available, I think using `build` for the conda package would have been fine. Certainly, the case where "build" shows up and works, but is stuck at an old version, is not at all ideal. I thought PyPI names were used wherever possible?

**Comment by zklaus** [comment]
I also don't feel to strongly about this, but ran into the same precise issue as you guys, i.e. installed `build` and was caught by it being out of date. There was some discussion and apparent consensus in https://github.com/conda-forge/build-feedstock/issues/2, but post-hoc some concerns surfaced.

I am not sure what is the best solution, but I agree that some solution would be desirable.

**Comment by zklaus** [comment]
This issue has lingered for a long time, but in the end it was decided to stick with the name `python-build` only, to not add a `build` alias here, and to mark all previous `build` packages (that at this point are really very out-of-date) as broken, c.f. https://github.com/conda-forge/admin-requests/pull/1048).

**Pull request closed without merging**